packages/template-retail-react-app/app/components/product-tile/index.jsx

```diff
@@ -66,15 +67,52 @@ const ProductTile = (props) => {
         ...rest
     } = props
 
-    const {currency, image, price, productId, hitType} = product
+    const {image, productId, hitType} = product
+
+    // NOTE: swatches will implement later to set variant accordingly,
+    // On first load, get the variant that is the represent product
+    // this is for variant product, standard/set/bundles does not have variants
+    // Also, product tile can be used in RecommendedProducts where it calls getProducts which does not have representedProduct
+    // in that case we use the first variant that has price book to set up discount price
+    // Not all variants has set in a priceBook, meaning not having tieredPrices.
+    const variant = product?.variants?.find(
+        (i) => i?.productId === product?.representedProduct?.id || !!i?.tieredPrices
+    )
```

Comment:
Is there a question about performance when there are large number of variations and a page is rendering a lot. Should we be doing this kind of looping in a `useMemo` perhaps?

Reply:
Can you explain the find function here? From the looks of it we are defining the variant as the object with a productId that is equal to the represented product id, which makes sense to me, but then we are saying it could also just be the first variant that has tiered prices. Are we trying to use that as a default? If so, is it possible that we run into the later condition before the first condition which is technically the better find return value?

Reply:
I wrote a note below to explain this behavior, represented_product only exists in productSearch result, we use this component for Recommendation section too, which use getProducts. 
When we can't use represented_product, we will have to go pick variants manually. However, not all variants has set in a priceBook, meaning not having tieredPrices.  For example, if you search for the product `classic wrap`, it is a variant product, but the first variant does not have a price book that we can apply like the rest of the variants. So I have to pick the first variant that has priceBook so we can get the tieredPrices here